Client/mods/deathmatch/logic/CClientPed.cpp

```diff
@@ -245,6 +245,12 @@ void CClientPed::Init(CClientManager* pManager, unsigned long ulModelID, bool bI
 
 CClientPed::~CClientPed(void)
 {
+    // A hack to destroy custom animation by playing a default internal animation.
+    // When IFP is unloaded by leaving the server, the pointer to its animations might
+    // still be somewhere in use, and a crash can occur by calling its members.
+    // So we switch to internal GTA animation to avoid the crash.
+    CStaticFunctionDefinitions::SetPedAnimation(*this, "ped", "idle_stance", -1, 250, true, false, false, false);
```

Comment:
Looks fine to me. What if this animation was replaced?

Reply:
Can't we just stop the current animation?

Reply:
`setPedAnimation` will play internal animations even if they are replaced, it's not affected by `engineReplaceAnimation`. 

@Jusonex We can only stop `TASK_PRIORITY_PRIMARY` animation i.e., animation played via `setPedAnimation` using `CClientPed::KillAnimation`. We cannot do the same for `TASK_PRIORITY_DEFAULT`, if we remove this task, the client will crash instantly. 

The main reason for crash is `CAnimBlendAssociation`, it is created for every ped when the animation changes, as it is not destroyed when you leave a server, GTA:SA thinks the animation is still running, so it keeps updating the bone positions for that animation. There are too many references to it in so many tasks, we'll need to do a lot of patching to safely destroy animation for every task, so this does the same thing, but it's easy and requires less code.